projects/vdk-plugins/vdk-test-utils/src/vdk/plugin/test_utils/util_plugins.py

```diff
@@ -57,7 +57,7 @@ def new_connection(self):
         import sqlite3
 
```

Comment:
It's not a python breaking change of the interface surface contract. It's a vdk-core one, for example:

* https://github.com/vmware/versatile-data-kit/blob/5b6ab858cda28f5374ec636cbf0cf8a9b63b6b5d/projects/vdk-core/src/vdk/api/data_job.py#L74 <- uses a python -depreacted and now -removed concept for method contract. The method signature should be refactored in a backwards-compatible way.
* https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/standalone_data_job.py#L148  <- it uses uses that concept for implementation, that should honour the refactored method signature.

Reply:
There's no breaking changes anywhere. What's happening is Python are changing their internal implementation structure. If they intend to change the public interfaces, they have not said so yet.

Reply:
It looks to me there is a change https://docs.python.org/3/library/sys.html#sys.exc_info
Is it test-covered, so we know for sure?

Reply:
Yes. there are test coverage. I checked just in case when you pointed out -test_run_standalone_data_job cover those scenarios.. 

But there's no incompatible change. There could not really be. Because it's a minor version of python. Python will not make non-backwards compatible changes in a minor version (at least not on purpose). They are following semantic versioning standard. Now it's possible some library to break (because they rely on non-standard things) but that's not the case here and our test cover this .

Reply:
Didn't quite find any coverage of the what seem to be now removed fields in Python 3.11.
Those are a part of the current VDK API https://github.com/vmware/versatile-data-kit/blob/a7951c0e6b0bf1e6cb4d2a28a319879c9be31a5f/projects/vdk-core/src/vdk/api/data_job.py#L74

The existing test I found is only partially covering the interface surface:
 https://github.com/vmware/versatile-data-kit/blob/a7951c0e6b0bf1e6cb4d2a28a319879c9be31a5f/projects/vdk-core/tests/functional/run/test_run_standalone_data_job.py#L65

Reply:
If there is breaking change, it will go through the code path of __exit__ with error case but since it passes with correct error, nothing is broken